slick/src/main/scala/slick/jdbc/MySQLProfile.scala

```diff
@@ -79,21 +79,24 @@ trait MySQLProfile extends JdbcProfile { profile =>
       override def name = super.name.filter(_ != "PRIMARY")
     }
     override def createColumnBuilder(tableBuilder: TableBuilder, meta: MColumn): ColumnBuilder = new ColumnBuilder(tableBuilder, meta) {
+      val BooleanPattern = "^([01])".r
+      val BitPattern = "^b'([01])'".r
       override def default = meta.columnDef.map((_,tpe)).collect{
         case (v,"String")    => Some(Some(v))
-        case ("1","Boolean") => Some(Some(true))
-        case ("0","Boolean") => Some(Some(false))
+        case ( BooleanPattern(v) , "Boolean") => Some( Some(v == "1"))
+        case ( BitPattern(v) , "Boolean") => Some( Some( v == "1"))
```

Comment:
may be simpler:
```
 case ( "0", "Boolean" ) => Some( Some(false) )
 case ( "1", "Boolean" ) => Some( Some(true) )
 case ( "b'0'", "Boolean") => Some( Some( false ) )
 case ( "b'1'", "Boolean") => Some( Some( true ) )
```

(and maybe even more correct because the regexes didn't use a trailing $, which mean BooleanPattern may have matched `12345` as true, not only 1).